wnfs-common/src/blockstore.rs

```diff
@@ -23,19 +24,26 @@ pub trait BlockStore {
     async fn put_block(&mut self, bytes: Vec<u8>, codec: IpldCodec) -> Result<Cid>;
 
     async fn put_serializable<V: Serialize>(&mut self, value: &V) -> Result<Cid> {
-        let bytes = dagcbor::encode(&ipld_serde::to_ipld(value)?)?;
+        let ipld = ipld_serde::to_ipld(value)?;
+
+        let mut bytes = Vec::new();
+        ipld.encode(DagCborCodec, &mut bytes)?;
+
         self.put_block(bytes, IpldCodec::DagCbor).await
     }
 
     async fn put_async_serializable<V: AsyncSerialize>(&mut self, value: &V) -> Result<Cid> {
         let ipld = value.async_serialize_ipld(self).await?;
-        let bytes = dagcbor::encode(&ipld)?;
+
+        let mut bytes = Vec::new();
+        ipld.encode(DagCborCodec, &mut bytes)?;
+
         self.put_block(bytes, IpldCodec::DagCbor).await
     }
 
-    async fn get_deserializable<V: DeserializeOwned>(&self, cid: &Cid) -> Result<V> {
+    async fn get_deserializable<'a, V: DeserializeOwned>(&'a self, cid: &Cid) -> Result<V> {
         let bytes = self.get_block(cid).await?;
-        let ipld = dagcbor::decode(bytes.as_ref())?;
+        let ipld = Ipld::decode(DagCborCodec, &mut Cursor::new(bytes.as_ref()))?;
         Ok(ipld_serde::from_ipld::<V>(ipld)?)
     }
 }
```

Comment:
This is undoing some changes that I think were good, presumably due to a merge conflict. Shouldn't we change them back?

Reply:
I suspect some of the rebases I did undid some of your changes. Tried to find all of them.